sklearn/tests/test_metaestimators_metadata_routing.py

```diff
@@ -171,6 +176,49 @@ def predict_log_proba(self, X, sample_weight="default", metadata="default"):
         # return np.zeros(shape=(len(X), 2))
 
 
+class ConsumingScorer(_BaseScorer):
+    def __init__(self, registry=None):
+        super().__init__(score_func="test", sign=1, kwargs={})
+        self.registry = registry
+
+    def __repr__(self):
+        return "Consuming_Scorer"
+
+    def __call__(
+        self, estimator, X, y_true, sample_weight="default", metadata="default"
+    ):
+        if self.registry is not None:
+            self.registry.append(self)
+
+        record_metadata_not_default(
+            self, "score", sample_weight=sample_weight, metadata=metadata
+        )
+
+        return 0.0
+
+
+class ConsumingSplitter(BaseCrossValidator, GroupsConsumerMixin):
+    def __init__(self, registry=None):
+        self.registry = registry
+
+    def split(self, X, y=None, groups="default"):
+        if self.registry is not None:
+            self.registry.append(self)
+
+        record_metadata_not_default(self, "split", groups=groups)
+
+        split_index = len(X) - 10
+        train_indices = range(0, split_index)
+        test_indices = range(split_index, len(X))
+        yield test_indices, train_indices
+
+    def get_n_splits(self, X=None, y=None, groups=None):
+        pass
```

Comment:
these should be either implemented (and used) or not present I think.

Reply:
TypeError: Can't instantiate abstract class ConsumingSplitter with abstract method get_n_splits

Reply:
I see, then you can mark the two `pass` lines with `# pragma: no cover`